src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -304,7 +304,9 @@ private function parseXmlAttributes(\DOMNode $node, array $context = [])
         $typeCastAttributes = $this->resolveXmlTypeCastAttributes($context);
 
         foreach ($node->attributes as $attr) {
-            if (!is_numeric($attr->nodeValue) || !$typeCastAttributes || (isset($attr->nodeValue[1]) && '0' === $attr->nodeValue[0])) {
+            if (!is_numeric($attr->nodeValue) ||
+                !$typeCastAttributes ||
+                (isset($attr->nodeValue[1]) && '0' === $attr->nodeValue[0] && '.' !== $attr->nodeValue[1])) {
```

Comment:
Will this also work for XML like this?

```xml
<document index="0">Name</document>
```

Reply:
I'd assume so, there is `isset($attr->nodeValue[1])` check

Reply:
Oh. I missed that. Thank you